src/OSX/Avalonia.MonoMac/ScreenImpl.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Avalonia.Platform;
+using MonoMac.AppKit;
+using MonoMac.Foundation;
+
+namespace Avalonia.MonoMac
+{
+    public class ScreenImpl : IScreenImpl
+    {
+        private const string NSApplicationDidChangeScreenParametersNotification = "NSApplicationDidChangeScreenParametersNotification";
+     
+        public int ScreenCount
+        {
+            get => NSScreen.Screens.Length;
+        }
+
+        public Screen[] AllScreens
+        {
+            get
+            {
+                if (_allScreens == null)
+                {
+                    NSScreen[] screens = NSScreen.Screens;
+                    Screen[] s = new Screen[screens.Length];
+                    NSScreen primary = NSScreen.MainScreen;
+                    for (int i = 0; i < screens.Length; i++)
+                    {
+                        Rect bounds = screens[i].Frame.ToAvaloniaRect().ConvertRectY();
+                        Rect workArea = screens[i].VisibleFrame.ToAvaloniaRect().ConvertRectY();
+                        s[i] = new MacScreen(bounds, workArea, screens[i] == primary, screens[i].Handle);
+                    }
+
+                    _allScreens = s;
+                    _observer = NSNotificationCenter.DefaultCenter.AddObserver(NSApplicationDidChangeScreenParametersNotification, notification =>
```

Comment:
I think this subscription should happen in static constructor instead. Same with GTK implementation, since neither GdkDisplay of GdkScreen would ever change.